Documentation/tutorial/howto_create_custom_plugin.md

```diff
@@ -0,0 +1,160 @@
+    }
+    ```
+
+    Here we declare this processor can handle any `.rtf` file in article category with normal priority.
+    When two or more processors declare for same file, DocFX will give it to the higher priority one.
+
+3.  Load our rtf file by read all text:
+    ```c#
+    public FileModel Load(FileAndType file, ImmutableDictionary<string, object> metadata)
+    {
+        var content = new Dictionary<string, object>
+        {
+            ["conceptual"] = File.ReadAllText(Path.Combine(file.BaseDir, file.File)),
+            ["type"] = "Conceptual",
+            ["path"] = file.File,
+        };
+        return new FileModel(file, content);
+    }
+    ```
+    Since we want to threat `.rtf` file as markdown conceptual file, so we use same data structure with markdown conceptual - `Dictionary<string, object>`.
+
+4.  Implements `Save` method as following:
+    ```c#
+    public override SaveResult Save(FileModel model)
+    {
+        return new SaveResult
+        {
+            DocumentType = "Conceptual",
+            ModelFile = model.File,
+        };
+    }
+    ```
+
+5.  Keep `UpdateHref` method empty.
+
+6.  `Name` property is used to display in log, so give any constant string as you like.
+    e.g.:
+    ```c#
+    public string Name => nameof(RtfDocumentProcessor);
+    ```
+
+7.  `BuildSteps` property is next important property, we suggest implement as following:
+    ```c#
+    [ImportMany(nameof(RtfDocumentProcessor))]
+    public IEnumerable<IDocumentBuildStep> BuildSteps { get; set; }
+    ```
+
+Create a document build step
+----------------------------
+1.  Create a new class (RtfBuildStep.cs), and declare it is for `RtfDocumentProcessor`:
+    ```c#
+    [Export(nameof(RtfDocumentProcessor), typeof(IDocumentBuildStep))]
+    public class RtfBuildStep : IDocumentBuildStep
+    {
+        // todo : implements IDocumentBuildStep.
+    }
+    ```
+
+2.  In `Build` method, convert rtf to html:
+    ```c#
+    private readonly TaskFactory _taskFactory = new TaskFactory(new StaTaskScheduler(1));
+
+    public void Build(FileModel model, IHostService host)
+    {
+        string content = (string)((Dictionary<string, object>)model.Content)["conceptual"];
+        content = _taskFactory.StartNew(() => RtfToHtmlConverter.ConvertRtfToHtml(content)).Result;
+        ((Dictionary<string, object>)model.Content)["conceptual"] = content;
+    }
+    ```
+
+3.  Implements other methods:
+    ```c#
+    public int BuildOrder => 0;
+
+    public string Name => nameof(RtfBuildStep);
+
+    public void Postbuild(ImmutableList<FileModel> models, IHostService host)
+    {
+    }
+
+    public IEnumerable<FileModel> Prebuild(ImmutableList<FileModel> models, IHostService host)
+    {
+        return models;
+    }
+    ```
+
+Enable plug-in
+--------------
+1.  Build our project.
+2.  Copy the output dll file.
+3.  Open DocFX.exe folder.
+4.  Create a folder with name `Plugins`.
+5.  Paste our dll file.
+
```

Comment:
We can also create a folder with `plugins` subfolder, and add this folder to `template` section in `docfx.json` to enable plugin